asciidoc-confluence-publisher-client/src/test/java/org/sahli/asciidoc/confluence/publisher/client/ConfluencePublisherTest.java

```diff
@@ -283,49 +287,182 @@ public void publish_metadataWithExistingPageWithSameContentButDifferentTitleAndR
+
+        // act
+        confluencePublisher.publish();
+
+        // assert
+        verify(confluenceRestClientMock).deleteAttachment("att1");
+        verify(confluenceRestClientMock).deletePropertyByKey("72189173", "attachmentOne.txt-hash");
+
+        verify(confluenceRestClientMock).deleteAttachment("att2");
+        verify(confluenceRestClientMock).deletePropertyByKey("72189173", "attachmentTwo.txt-hash");
+    }
+
+    @Test
+    public void publish_whenOldAttachmentsStillExist_deletesAttachmentsNotPresentUnderPage() {
+        // arrange
+        ConfluenceRestClient confluenceRestClientMock = mock(ConfluenceRestClient.class);
+        when(confluenceRestClientMock.getPageWithContentAndVersionById("72189173")).thenReturn(new ConfluencePage("72189173", "Existing Page (Old Title)", "<h1>Some Confluence Content</h1>", 1));
+        when(confluenceRestClientMock.getPropertyByKey("72189173", CONTENT_HASH_PROPERTY_KEY)).thenReturn(SOME_CONFLUENCE_CONTENT_SHA256_HASH);
+
+        when(confluenceRestClientMock.getAttachments("72189173")).thenReturn(asList(
+                new ConfluenceAttachment("att1", "attachmentOne.txt", "", 1),
+                new ConfluenceAttachment("att2", "attachmentTwo.txt", "", 1),
+                new ConfluenceAttachment("att3", "attachmentThree.txt", "", 1)
+
+        ));
+
+        when(confluenceRestClientMock.getAttachmentByFileName("72189173", "attachmentOne.txt")).thenReturn(new ConfluenceAttachment("att1", "attachmentOne.txt", "", 1));
+        when(confluenceRestClientMock.getPropertyByKey("72189173", "attachmentOne.txt-hash")).thenReturn(sha256Hex("attachment1"));
+
+        when(confluenceRestClientMock.getAttachmentByFileName("72189173", "attachmentTwo.txt")).thenReturn(new ConfluenceAttachment("att2", "attachmentTwo.txt", "", 1));
+        when(confluenceRestClientMock.getPropertyByKey("72189173", "attachmentTwo.txt-hash")).thenReturn(sha256Hex("attachment2"));
+
+        ConfluencePublisher confluencePublisher = confluencePublisher("root-ancestor-id-page-with-attachments", REPLACE_ANCESTOR, confluenceRestClientMock, null, null);
+
+        // act
+        confluencePublisher.publish();
+
+        // assert
+        verify(confluenceRestClientMock, never()).deleteAttachment("att1");
+        verify(confluenceRestClientMock, never()).deletePropertyByKey("72189173", "attachmentOne.txt-hash");
+
+        verify(confluenceRestClientMock, never()).deleteAttachment("att2");
+        verify(confluenceRestClientMock, never()).deletePropertyByKey("72189173", "attachmentTwo.txt-hash");
+
+        verify(confluenceRestClientMock).deleteAttachment("att3");
+        verify(confluenceRestClientMock).deletePropertyByKey("72189173", "attachmentThree.txt-hash");
+    }
+
     @Test
     public void publish_metadataWithExistingPageAndAttachmentWithDifferentAttachmentContentUnderRootSpace_sendsUpdateAttachmentRequest() {
```

Comment:
Is this scenario not already covered by `publish_whenOldAttachmentsStillExist_deletesAttachmentsNotPresentUnderPage` ?